service/r2/routes.go

```diff
@@ -64,6 +64,27 @@ func createNamespace(s *service, r *http.Request) (data interface{}, err error)
 	return newNamespaceJSON(view), nil
 }
 
+func validateRuleSet(s *service, r *http.Request) (data interface{}, err error) {
+	vars := mux.Vars(r)
+	var rsj ruleSetJSON
+	if err := parseRequest(&rsj, r.Body); err != nil {
+		return nil, err
+	}
+	if vars[namespaceIDVar] != rsj.Namespace {
+		return nil, fmt.Errorf(
+			"namespaceID param %s and ruleset namespaceID %s do not match",
+			vars[namespaceIDVar],
+			rsj.Namespace,
+		)
+	}
+
+	rss := rsj.ruleSetSnapshot(ruleSetSnapshotOpts{generateMissingID: true})
+	if err := s.store.ValidateRuleSet(rss); err != nil {
+		return nil, err
+	}
+	return fmt.Sprintf("Ruleset is valid"), nil
```

Comment:
1) I think you can just return a string, not need to use fmt.Sprintf

2) Given that this just returns a string that tells the caller that seems a little self-explanatory do you think this function could just return an error?

Reply:
1) Thanks - nice catch.
2) Following convention on the other endpoints that return string values in the service (ie deleteMappingRule, deleteRollupRule, etc.) I'm open to whatever either way.